### PR TITLE
test(openchallenges): improve organization service model entity unit test coverage

### PR DESCRIPTION
**Pull request by mdsage1**
## Description

Sonar unit organization service test coverage has been increased to 45% for model/entity package by writing unit tests for a single class, OrganizationEntity.

## Related Issue

#2063 
#1528 
#2348 
#2388 

## Changelog

See `files Changed` section

These changes have increased the Sonar coverage from the 15% achieved with the exception unit test PR #2348 (9.45% overall). These tests were written last week and debugged this week. The PR has been released after the approval of the exception unit test PR on 2023-12-04.

## Preview

Overall sonar coverage has increased to 50%, from 0%, for organization service model entity.

The Sonar coverage increase for model/mapper:
image

The goal was to get organization-service > 30% and I will confirm this once the PR has been merged. Given that the Sonar coverage calculation is 15% higher than the intended coverage it's possible that the [Code Quality Dashboard](https://github.com/Sage-Bionetworks/sage-monorepo/blob/main/CODE_QUALITY.md) will reflect 30% code coverage with the updates made to this class.

### CODE REVIEW

**Comment by mdsage1** [comment]
@tschaffter Do you feel this covers #1528? Or were you thinking about another class when creating this?

File: apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+            .websiteUrl(websiteUrl)
+            .challengeCount(challengeCount)
+            .categories(Collections.emptyList())
+            .challengeContributions(Collections.emptyList())
+            .description(description)
+            .createdAt(createdAt)
+            .updatedAt(updatedAt)
+            .acronym(acronym)
+            .build();
+    Long retrievedId = organizationEntity.getId();
```
Comment (tschaffter) [inline apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java:43]: Two name terms used often together are "actual" and "expected" value. Here you could replace "retrieved" by "actual". You could optionally define above "expectedName", etc. but it's fine with me to keep using "name", "email", etc.

As usual, please consider other places where this comment may apply (I may occasionally no longer mention this).

File: apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    organizationEntity2.setName("Test Organization");
+    organizationEntity2.setEmail("test@example.com");
+    organizationEntity2.setLogin("testorg");
+    organizationEntity2.setDescription("Test description");
+    organizationEntity2.setAvatarKey("avatarKey");
+    organizationEntity2.setWebsiteUrl("https://example.com");
+    organizationEntity2.setChallengeCount(5);
+    organizationEntity2.setCreatedAt(createdAt);
+    organizationEntity2.setUpdatedAt(updatedAt);
+    organizationEntity2.setAcronym("TO");
```
Comment (tschaffter) [inline apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java:166]: This is a good opportunity to start using a "@BeforeAll" (see this [article](https://www.baeldung.com/junit-before-beforeclass-beforeeach-beforeall)) to define objects ONCE and then use them in more than one test.

**Review by tschaffter — CHANGES_REQUESTED** [review]

**Comment by mdsage1** [comment]
Changes implemented:
1. Updated organizationEntity to entity in tests with a single entity 
2. Changed all occurrences of `retrieved` in variable names to `actual`
3. TBD on how to implement @BeforeAll w/o throwing errors

File: apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java (hunk 2 above)
Comment (mdsage1) [reply apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java]: Added `@BeforeAll` to the code to initialized the properties of each organization entity that is generated.

File: apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    entityFromConstructor2.setCreatedAt(createdAt);
+    entityFromConstructor2.setUpdatedAt(updatedAt);
+    entityFromConstructor2.setAcronym(acronym);
+    entityFromConstructor2.setCategories(Collections.emptyList());
+    entityFromConstructor2.setChallengeContributions(Collections.emptyList());
+  }
+
+  @Test
+  public void
+      OrganizationEntityBuilderArguments_ShouldReturnArgumentsPassed_WhenOrganizationEntityBuilt() {
```
Comment (tschaffter) [inline apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java:74]: Review title: the "BuilderArguments" don't "return" anything. You could say that the org entity returns the arguments passed when built with the class builder.

**Review by tschaffter — CHANGES_REQUESTED** [review]

File: apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+import java.time.OffsetDateTime;
+import java.util.Collections;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class OrganizationEntityTest {
+
+  private static Long id;
```
[the remark this answers is not in the document]
Comment (mdsage1) [reply apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java:11]: addressed in the `Define these values at the same time you define the class member variables for a more concise code.` change request.

File: apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    name = "Test Organization";
+    email = "test@example.com";
+    login = "testorg";
+    avatarKey = "avatarKey";
+    websiteUrl = "https://example.com";
+    challengeCount = 5;
+    description = "Test description";
+    createdAt = OffsetDateTime.now();
+    updatedAt = OffsetDateTime.now();
+    acronym = "TO";
```
[the remark this answers is not in the document]
Comment (mdsage1) [reply apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java:37]: defined the values at the same time as defining the class variables and removed `static`.

File: apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java (hunk 3 above)
Comment (mdsage1) [reply apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java:63]: Changed the name to `OrganizationEntity_ShouldReturnArgumentsPassed_WhenOrganizationEntityBuiltWithClassBuilder`

File: apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+    Assertions.assertEquals(
+        Collections.emptyList(), entityFromConstructor.getChallengeContributions());
+    Assertions.assertEquals(description, entityFromConstructor.getDescription());
+    Assertions.assertEquals(createdAt, entityFromConstructor.getCreatedAt());
+    Assertions.assertEquals(updatedAt, entityFromConstructor.getUpdatedAt());
+    Assertions.assertEquals(acronym, entityFromConstructor.getAcronym());
+  }
+
+  @Test
+  public void HashCode_ShouldBuildHashCodeOfArgumentsForOrganizationEntity_WhenArgumentsPassed() {
```
[the remark this answers is not in the document]
Comment (mdsage1) [reply apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java:118]: Changed to `HashCode_ShouldBuildHashCodeOfOrganizationEntity_WhenArgumentsPassed`

File: apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+    Assertions.assertEquals(Collections.emptyList(), entity.getChallengeContributions());
+    Assertions.assertEquals(description, entity.getDescription());
+    Assertions.assertEquals(createdAt, entity.getCreatedAt());
+    Assertions.assertEquals(updatedAt, entity.getUpdatedAt());
+    Assertions.assertEquals(acronym, entity.getAcronym());
+  }
+
+  @Test
+  public void
+      OrganizationEntityMethodsToSetArguments_ShouldReturnArguments_WhenOrganizationEntityArgumentsPassed() {
```
[the remark this answers is not in the document]
Comment (mdsage1) [reply apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java:99]: Changed the name to `OrganizationEntityGetters_ShouldReturnExpectedValues_WhenArgumentsPassed`

**Comment by mdsage1** [comment]
Organization entity properties, login and email, were removed during PR #2399 which caused the OrganizationEntityTest.java unit tests to fail. Removed the unit tests. Sonar coverage was reduced to 48%.
image

File: apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+  public void HashCode_ShouldBuildHashCodeOfOrganizationEntity_WhenArgumentsPassed() {
+
+    int hashCode1 = entityFromConstructor.hashCode();
+    int hashCode2 = entityFromConstructor2.hashCode();
+
+    Assertions.assertEquals(hashCode1, hashCode2);
+  }
+
+  @Test
+  public void EqualsObject_ShouldReturnBoolean_WhenPassedTwoObjects() {
```
[the remark this answers is not in the document]
Comment (mdsage1) [reply apps/openchallenges/organization-service/src/test/java/org/sagebionetworks/openchallenges/organization/service/model/entity/OrganizationEntityTest.java]: Implemented the following changes:

1. Changed the unit test name from EqualsObject_ShouldReturnBoolean_WhenPassedTwoObjects to EqualsObject_ShouldBeTheSame_WhenPassedTwoIdenticalObjects
2. Separated that test into 2 smaller tests 1 test differentiates between two different objects and the other recognizes two objects as being identical.

**Comment by mdsage1** [comment]
@tschaffter This is ready for you to review. I made the changes you asked for:

> Implemented the following changes:
> 
> 1. Changed the unit test name from EqualsObject_ShouldReturnBoolean_WhenPassedTwoObjects to EqualsObject_ShouldBeTheSame_WhenPassedTwoIdenticalObjects
> 2. Separated that test into 2 smaller tests 1 test differentiates between two different objects and the other recognizes two objects as being identical.

**Pull request merged**